src/meshapi/models/member.py

```diff
@@ -4,8 +4,21 @@
 from django.db.models.fields import EmailField
 from django_jsonform.models.fields import ArrayField as JSONFormArrayField
 
+class ContactPreference(models.TextChoices):
+    EMAIL = 'EMAIL', 'Email'
+    PHONE = 'PHONE', 'Phone'
+    SLACK = 'SLACK', 'Slack'
+    OTHER = 'OTHER', 'Other'
```

Comment:
I don't think we should provide Slack as an option here, since we don't currently have a way to collect slack handles (and this really doesn't make sense at join form time). I also think we should use null DB values rather than Other, since this more appropriately captures the unknown-ness. "Other" makes it sound like they have explicitly told us they don't want email or phone